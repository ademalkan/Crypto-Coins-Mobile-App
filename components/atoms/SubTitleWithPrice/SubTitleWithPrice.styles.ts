import { StyleSheet } from "react-native";

export default StyleSheet.create({
  container: {
    display: "flex",
    flexDirection: "row",
  },
  subtitle: {
    color: '#666',
    fontWeight: "bold",
    fontSize: 14,
    marginBottom: 4,
    textTransform: 'capitalize',
  },
  price: {
    color: '#666',
    fontSize: 14,
    fontWeight: '500',
    marginLeft: 4
  },
})