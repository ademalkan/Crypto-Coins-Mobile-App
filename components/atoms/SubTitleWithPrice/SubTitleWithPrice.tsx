import React from 'react';
import { View, Text } from 'react-native';
import styles from './SubTitleWithPrice.styles'
interface SubTitleWithPriceProps {
  text: string;
  price: number;
}

const SubTitleWithPrice: React.FC<SubTitleWithPriceProps> = ({ text, price }) => {
  return (
    <View style={styles.container}>
      <Text style={styles.subtitle}>{text}</Text>
      <Text style={styles.price}>${price?.toFixed(2)}</Text>
    </View>
  );
};

export default SubTitleWithPrice;
