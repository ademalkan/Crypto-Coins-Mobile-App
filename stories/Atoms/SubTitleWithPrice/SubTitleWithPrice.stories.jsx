import React from "react";
import { View } from "react-native";
import SubTitleWithPrice from "../../../components/atoms/SubTitleWithPrice/SubTitleWithPrice";

export default {
  title: "Atoms/SubTitleWithPrice",
  component: SubTitleWithPrice,
};

const subtitle = "Subtitle Text";
const price = 99.99;

export const Default = () => (
  <View style={{ alignItems: "center", justifyContent: "center", flex: 1 }}>
    <SubTitleWithPrice text={subtitle} price={price} />
  </View>
);
