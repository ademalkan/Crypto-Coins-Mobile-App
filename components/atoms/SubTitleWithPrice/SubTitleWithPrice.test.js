import React from "react";
import renderer from "react-test-renderer";
import { View, Text } from "react-native";
import SubTitleWithPrice from "./SubTitleWithPrice";

describe("SubTitleWithPrice", () => {
  it("should render the text and price props correctly", () => {
    const text = "Subtitle Text";
    const price = 9.99;

    const component = renderer.create(
      <SubTitleWithPrice text={text} price={price} />
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
