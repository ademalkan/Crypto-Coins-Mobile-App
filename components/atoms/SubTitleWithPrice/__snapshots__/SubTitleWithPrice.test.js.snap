// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubTitleWithPrice should render the text and price props correctly 1`] = `
<View
  style={
    {
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Text
    style={
      {
        "color": "#666",
        "fontSize": 14,
        "fontWeight": "bold",
        "marginBottom": 4,
        "textTransform": "capitalize",
      }
    }
  >
    Subtitle Text
  </Text>
  <Text
    style={
      {
        "color": "#666",
        "fontSize": 14,
        "fontWeight": "500",
        "marginLeft": 4,
      }
    }
  >
    $
    9.99
  </Text>
</View>
`;
